functions/__fzf_search_git_log.fish

```diff
@@ -1,14 +1,11 @@
 # Originally implemented in and transposed from https://github.com/patrickf3139/dotfiles/pull/2
 function __fzf_search_git_log --description "Search the git log of the current git repository. Insert the selected commit hash into the commandline at the cursor."
-    if not git rev-parse --git-dir >/dev/null 2>&1
+    # see documentation for git format placeholders at https://git-scm.com/docs/git-log#Documentation/git-log.txt-emnem
+    # %h gives you the abbreviated commit hash, which is useful for saving screen space, but we will have to expand it later below
+    if not set --local gitLog (git log --color=always --format=format:'%C(bold blue)%h%C(reset) - %C(cyan)%as%C(reset) %C(yellow)%d%C(reset) %C(normal)%s%C(reset)  %C(dim normal)[%an]%C(reset)' 2>/dev/null)
```

Comment:
One more thing, can you explain why you prefer `set --local` vs `set`?

Reply:
Because they are not the same thing. From the docs:

"If a variable is not explicitly set to be either universal, global or local and has never before been defined, the variable will be local to the currently executing function. Note that this is different from using the -l or --local flag. If one of those flags is used, the variable will be local to the most inner currently executing block, while without these the variable will be local to the function. If no function is executing, the variable will be global."

`set --local` is more clear, and is usually what you intend anyways. When there are exceptions, it's nice to make that clear by _not_ including the `--local`.

Reply:
When defining a variable at the top level in a function, `set` and `set --local` are the same. But clarity is good imo.